### PR TITLE
ref(poetry): `pipenv` -> `poetry`

### LINKED ISSUE #377: Switch dependency management/build systems from `pipenv` to either `poetry` or `pdm`
**Is your feature request related to a problem? Please describe.**
Pipenv doesn't have a build system, so we have to rely on a separate `setup.py` or `setup.cfg` for it (which kind of defeats the purpose of having a `Pipfile`). Also it's dependency resolution isn't the greatest. Whereas [poetry](https://python-poetry.org/) or [pdm](https://pdm.fming.dev/) have it, and are in general better tools for the job

**Describe the solution you'd like**
Switch to either `pdm` or `poetry`

**Describe alternatives you've considered**
Use any other system?

**Additional context**
Here's an article comparing the systems https://dev.to/frostming/a-review-pipenv-vs-poetry-vs-pdm-39b4

This is a part of #358. This will potentially close #368 and replace it with a PR for `pyproject.toml` (which is the "setup.py" here)

### PR DESCRIPTION
**Pull request by wizard-28**
## Purpose

Closes #377 

## Checklist

- [x] I confirm that I have read the [contributing guidelines](https://github.com/pacstall/pacstall/blob/develop/CONTRIBUTING.md), and this pull request is abiding by all the clauses stated in the guideline.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+[tool.poetry]
+name = "pacstall"
+version = "2.0"
```
Comment (D-Brox) [inline pyproject.toml:3]: Should we put '2.0-dev" for now?

Comment (wizard-28) [reply pyproject.toml:3]: Why?

Comment (Henryws) [reply pyproject.toml:3]: Because it’s a dev thingy I assume

Comment (wizard-28) [reply pyproject.toml:3]: Resolved in https://github.com/pacstall/pacstall/discussions/379/#discussioncomment-1689282

**Review by Henryws — APPROVED** [review]

**Review by Henryws — APPROVED** [review]

**Review by D-Brox — APPROVED** [review]

**Pull request merged**